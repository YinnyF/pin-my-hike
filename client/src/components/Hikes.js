/* eslint-disable no-unused-vars */
import React from "react";
import Form from './Form.js'

export const Hikes = () => {
  const [pins, setPins] = React.useState([]);
  const [addPins, setAddPins] = React.useState(false);

  React.useEffect(() => {
    const getPins = async () => {
      const pinsFromServer = await fetchPins()
      setPins(pinsFromServer)
    }
    getPins();
  }, []);

  React.useEffect(() => {
    fetchPins().then(u => setPins(u))
  }, [])

  const fetchPins = async () => {
    const res = await fetch('http://localhost:3002/pins')
    const pins = await res.json()

    return pins
  }

  const addPin = async (pin) => {
    const res = await fetch('http://localhost:3002/pins', { 
      method: 'POST', 
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(pin)
    })
    const data = await res.json()

    setAddPins([...addPins, data])
  }

  return (
    <div data-testid="hikes">
      
    </div>
  )
};
