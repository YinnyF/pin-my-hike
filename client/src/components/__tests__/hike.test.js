import React from "react";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Hikes } from "../Hikes";

afterEach(() => {
  cleanup();
});

test("should match the snapshot", () => {
  const tree = renderer.create(<Hikes />).toJSON();
  expect(tree).toMatchSnapshot();
});
