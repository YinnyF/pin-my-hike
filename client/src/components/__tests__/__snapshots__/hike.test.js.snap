// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  data-testid="hike"
>
  Hello from server!
  <h1
    className="hikes"
  />
</div>
`;
